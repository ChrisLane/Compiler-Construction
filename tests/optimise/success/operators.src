// This test should reduce operators with basic data types to their end value.
main() {
    var x = 1 + 1;
    x; // Don't optimise new
    x = 1 - 1;
    x = 1 * 1;
    x = 1 / 1;

    var y = 1 <= 1;
    y; // Don't optimise new
    y = 1 >= 1;
    y = 1 == 1;
    y = 1 != 1;

    var z = true && true;
    z; // Don't optimise new
    z = true || true;
    z = !true;
}
