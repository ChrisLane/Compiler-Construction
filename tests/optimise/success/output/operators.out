
main (  ) { 
    New ( "x" = Const 2 ) In { 
        Identifier "x" ; 
        Asg ( Identifier "x" := Const 0 ) ; 
        Asg ( Identifier "x" := Const 1 ) ; 
        Asg ( Identifier "x" := Const 1 ) ; 
        New ( "y" = Bool true ) In { 
            Identifier "y" ; 
            Asg ( Identifier "y" := Bool true ) ; 
            Asg ( Identifier "y" := Bool true ) ; 
            Asg ( Identifier "y" := Bool false ) ; 
            New ( "z" = Bool true ) In { 
                Identifier "z" ; 
                Asg ( Identifier "z" := Bool true ) ; 
                Asg ( Identifier "z" := Bool false ) 
            } 
        } 
    } 
}

